session/session.go

```diff
@@ -1369,6 +1369,11 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	if err := executor.ResetContextOfStmt(s, stmtNode); err != nil {
 		return nil, err
 	}
+	if err := s.validateStatementReadOnlyInStaleness(stmtNode); err != nil {
+		// TODO: Do we need to rollback txn?
+		s.RollbackTxn(ctx)
+		return nil, err
+	}
```

Comment:
If we detact any write statement during read-only staleness transaction, do we need to rollback the transaction? @youjiali1995

Reply:
No. Reporting an error is enough.

Reply:
updated. @djshow832 @nolouch